src/common/Thread.cc

```diff
@@ -31,6 +33,9 @@
 #include <sched.h>
 #endif
 
+// STATIC
+std::mutex Thread::threadid_name_mutex;
```

Comment:
While this is just fine, you did mention that a using a spinlock would suffice. Did that not work out for some reason?